.circleci/config.yml

```diff
@@ -113,6 +92,115 @@ jobs:
       - EVENT_LOOP: asyncio
       - TIMEOUT: "6.0"
 
+  lint:
+    docker:
+      - image: saltyrtc/circleci-image-python:python-3.7
+    steps:
+      - checkout
+      - run:
+          name: Install SaltyRTC Server
+          command: pip install .[dev]
+      - run:
+          name: Run Flake8
+          command: flake8 .
+      - run:
+          name: Run isort
+          command: isort -rc -c . || (isort -rc -df . && exit 1)
+      - run:
+          name: Run Mypy
+          command: MYPYPATH=${PWD}/stubs mypy saltyrtc examples
+      - run:
+          name: Run checkdocs
+          command: python setup.py checkdocs
+
+  build-docker:
+    docker:
+      - image: docker:stable
+    parameters:
+      rebuild:
+        type: boolean
+        default: false
+      supported-tags:
+        type: string
+        default: "^v4\\.[1-9]+\\.[0-9]+$"
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Determine tag name
+          command: |
+            tag=${CIRCLE_BRANCH}
+            if [ "${CIRCLE_BRANCH}" = "master" ]; then
+              tag="latest"
+            elif [[ "${CIRCLE_BRANCH}" =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+              tag=${CIRCLE_BRANCH:1}
+            fi
+            echo "Mapping branch ${CIRCLE_BRANCH} to ${tag}"
+            echo "export TAG=${tag}" >> ${BASH_ENV}
+      - run:
+          name: Build Docker image
+          command: |
+            docker build \
+              --no-cache \
+              -t saltyrtc/saltyrtc-server-python:${TAG} \
+              .
+      - run:
+          name: Push Docker image
+          command: |
+            docker login -u ${DOCKER_USER} -p ${DOCKER_API_KEY}
+            docker push saltyrtc/saltyrtc-server-python:${TAG}
+      - when:
+          condition: << parameters.rebuild >>
+          steps:
+            - run:
+                name: Determine supported tags
+                command: |
+                  for pattern in "<< parameters.supported-tags >>"; do
+                    tags=$(git tag -l --sort=-v:refname | grep -E $pattern | tr '\n' ' ')
```

Comment:
It would probably be good to enclose the `$pattern` in double quotes, otherwise if I'm not mistaken a pattern containing spaces would fail.

Reply:
Fine by me. Btw. that's your code, so you may want to adjust.

Reply:
But then again, it can't contain spaces.

Reply:
Did I mention that I hate bash?